pkg/apis/kops/validation/legacy.go

```diff
@@ -259,27 +259,45 @@ func ValidateCluster(c *kops.Cluster, strict bool) *field.Error {
 		}
 	}
 
-	// Check KubeDNS.ServerIP
+	// @check the custom kubedns options are valid
 	if c.Spec.KubeDNS != nil {
-		serverIPString := c.Spec.KubeDNS.ServerIP
-		if serverIPString == "" {
-			return field.Required(fieldSpec.Child("KubeDNS", "ServerIP"), "Cluster did not have KubeDNS.ServerIP set")
-		}
 
-		dnsServiceIP := net.ParseIP(serverIPString)
-		if dnsServiceIP == nil {
-			return field.Invalid(fieldSpec.Child("KubeDNS", "ServerIP"), serverIPString, "Cluster had an invalid KubeDNS.ServerIP")
+		if c.Spec.KubeDNS.ServerIP != "" {
```

Comment:
This looks like it is making ServerIP optional?

Reply:
Yeah you need to error out, the same as in ServerIP

Reply:
so i figured i'd make it optional as if not set the server ip is later filled in by the [model](https://github.com/kubernetes/kops/pull/4724/files#diff-75a63eeab8d3d0d8b711a73414c50e4dR49) ...The user experience being, I want to change the image, replicas, or in this case the upstream server .. but i'm happy with the server ip